lib/inspector.rb

```diff
@@ -89,4 +89,30 @@ def scope
   def scope
     self.class.scope
   end
+
+  # Runs the given script on the inspected machine asynchronously and calls the callback method
+  # periodically with new output when it occurs.
+  #
+  # Example:
+  #
+  #     count = 0
+  #     raw_list = run_inspect_script("changed_managed_files.sh") do |chunk|
+  #       count += chunk.lines.count
+  #       Machinery::Ui.progress("Found #{count} changed files...")
+  #     end
+  def run_inspect_script(*script, &callback)
```

Comment:
What about calling the method `run_script_with_progress`? That would make it clearer why there is an additional method.